packages/treasury/src/vault.js

```diff
@@ -167,20 +169,24 @@ export function makeVaultKit(
     assert(AmountMath.isGTE(runReturned, runDebt));
 
     // Return any overpayment
-    vaultSeat.incrementBy(seat.decrementBy({ RUN: runDebt }));
+
+    const { zcfSeat: burnSeat } = zcf.makeEmptySeatKit();
+    burnSeat.incrementBy(seat.decrementBy({ RUN: runDebt }));
     seat.incrementBy(
       vaultSeat.decrementBy({ Collateral: getCollateralAllocated(vaultSeat) }),
     );
-    zcf.reallocate(seat, vaultSeat);
-
+    zcf.reallocate(seat, vaultSeat, burnSeat);
+    runMint.burnLosses({ RUN: runDebt }, burnSeat);
     seat.exit();
+    burnSeat.exit();
     active = false;
     updateUiState();
 
-    runMint.burnLosses({ RUN: runDebt }, vaultSeat);
     runDebt = AmountMath.makeEmpty(runBrand);
     assertVaultHoldsNoRun();
     vaultSeat.exit();
```

Comment:
This should have been exited when we liquidated the vault. Otherwise there's a race where more collateral could be added, for example.

Reply:
We now exit when liquidation starts. It's also needed here in close.